lua/neorg/modules/core/summary/module.lua

```diff
@@ -40,16 +40,54 @@ module.load = function()
     local ts = module.required["core.integrations.treesitter"]
 
     module.config.public.strategy = neorg.lib.match(module.config.public.strategy)({
-        metadata = function()
-            return function(files, heading_level)
+        default = function()
+            -- declare query on load so that it's parsed once, on first use
+            local heading_query
+
+            local get_first_heading_title = function(bufnr)
+                local document_root = ts.get_document_root(bufnr)
+                if not heading_query then
+                    -- allow second level headings, just in case
+                    local heading_query_string = [[
+                         [
+                             (heading1
+                                 title: (paragraph_segment) @next-segment
+                             )
+                             (heading2
+                                 title: (paragraph_segment) @next-segment
+                             )
+                         ]
+                     ]]
+                    heading_query = neorg.utils.ts_parse_query("norg", heading_query_string)
+                end
+                -- search up to 20 lines (a doc could potentially have metadata without metadata.title)
+                for _, heading in heading_query:iter_captures(document_root, bufnr, 1, 20) do
```

Comment:
Check between lines `1` and `20` is alright, but I think we might be better off altogether by just checking for the first occurrence in the file? I can imagine some niche scenarios where you have a long introductory paragraph to a document and only then do you have the title in a first level heading >20 lines down.

If you agree this is the way to go then the simplest way is to ditch the `for` loop altogether and invoke the function manually to get the first result like this:

```suggestion
				local _, heading = heading_query:iter_captures(document_root, bufnr)()
                
                if not heading then
                	-- do something
                end
                
                -- continue using the `heading` node normally
```

Reply:
done - thanks for the tip, I didn't know about using `(..)()` to iterate over the first item only.